MapboxCoreNavigation/MMEEventsManager.swift

```diff
@@ -164,6 +182,12 @@ extension MMEEventsManager {
         return eventDictionary
     }
     
+    func navigationFoundFasterRoute(routeController: RouteController) -> [String: Any] {
+        var eventDictionary = self.addDefaultEvents(routeController: routeController)
+        eventDictionary["event"] = FasterRouteFoundEvent
```

Comment:
We should also add `step`, `newDistanceRemaining`, `newDurationRemaining`, and `newGeometry` as we do in the reroute event.

Reply:
We also may want to have this leverage the `RerouteEvent.update` function: https://github.com/mapbox/mapbox-navigation-ios/blob/10ff07f2baff2574595a92ba425de63a4939a022/MapboxCoreNavigation/MMEEventsManager.swift#L381-L389